drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/RulesSet.java

```diff
@@ -48,7 +48,7 @@ public Model toExecModel(RulesExecutionController rulesExecutionController) {
         ModelImpl model = new ModelImpl();
         rules.stream().map(RuleContainer::getRule)
                 .map(r -> r.withOptions(options))
-                .flatMap(rule -> rule.toExecModelRules(this, rulesExecutionController, ruleCounter).stream())
+                .flatMap(rule -> rule.ruleGenerationContext.toExecModelRules(this, rule, rulesExecutionController, ruleCounter).stream())
```

Comment:
Keep the `toExecModelRules` on the `rule` and performs the call to the `ruleGenerationContext` from inside that method.